PlayFabClientSDK/source/PluginManager.cs

```diff
@@ -1,44 +1,100 @@
+using System;
 using System.Collections.Generic;
+using PlayFab.Internal;
+using PlayFab.Json;
 
 namespace PlayFab
 {
     public class PluginManager
     {
-        private Dictionary<PluginContract, IPlayFabPlugin> plugins = new Dictionary<PluginContract, IPlayFabPlugin>();
+        private Dictionary<Tuple<PluginContract, string>, IPlayFabPlugin> plugins = new Dictionary<Tuple<PluginContract, string>, IPlayFabPlugin>();
 
         /// <summary>
         /// The singleton instance of plugin manager.
         /// </summary>
-        public static PluginManager Instance { get; private set; } = new PluginManager();
+        private static PluginManager Instance { get; set; } = new PluginManager();
```

Comment:
Wow... this is ...
You have taught me something new.
In the meantime: Why is this a property instead of a field?